### ISSUE TITLE
Any plan to support carthge --use-xcframeworks?

### ISSUE THREAD

**Issue by xuzhaocheng** [SVGKit/SVGKit#736]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/SVGKit/SVGKit/blob/2.x/CONTRIBUTING.md)
* [x] I have reproduced the problem while running in Xcode, and read the Console log, and acted upon any relevant error or warning messages
* [x] I have reproduced the problem while running in Xcode's Debugger, and used breakpoints to examine the actual values of variables vs the expected values
* [x] I am not using CocoaPods,  I am using SVGKit ~> 2.1.0

### Issue Description and Steps

Since after carthage 0.37, we are recommended to use xcframworks rather than a single binary. Moreover, if we support xcframeworks, we could also support Apple Silicon.

**Comment by renep** [comment]
I have created a fork and updated the project that xcframeworks can be build using carthage. I use this in one of my iOS project.
Because there are some project changes that would break backwards compatibility with this project I have not yet created a pull request.
If anybody is interested in the fork you can find it here: https://github.com/renep/SVGKit/tree/xcframework

**Comment by adamgit** [comment]
If I understand correctly: this is an overall improvement we should push ahead with.

Can you summarise the changes to backwards compatibility?

If they're big enough, but the advantage is enough, could be worth us moving to v4.x?

**Comment by renep** [comment]
What I have changed is that the Cocoalumberjack dependency is now included using Carthage and references not the xcframework.
I have removed the static library build, because I had build errors and I don't need these. I'm wondering if anybody is using static libs nowadays where you have the option by using Cocoapods, Carthage or SwiftPM. If anybody wants to add a binary framework directly, then a xcframework binary build be the way to go.

In my fork the iOS Demo app works, but the macOS Version needs some update, because it does not compile. Also I don't think that 3 different targets for iOS, macOS and tvOS are needed. This is the reason xcframeworks exists.
I also have not tested if Cocoapods of SwiftPM works.

**Comment by adamgit** [comment]
Yeah, that sounds like the right way to go - SVGKit predates xcframeworks
by many years, and we never wanted to do static builds (But Apple forced us
to - they actively blocked us from using frameworks, refused to have them
on app store), so the static builds are a legacy workaround.

If someone can get this to build on the other platforms too then we could
have a really nice cleanup.

**Comment by kjy95** [comment]
> What I have changed is that the Cocoalumberjack dependency is now included using Carthage and references not the xcframework. I have removed the static library build, because I had build errors and I don't need these. I'm wondering if anybody is using static libs nowadays where you have the option by using Cocoapods, Carthage or SwiftPM. If anybody wants to add a binary framework directly, then a xcframework binary build be the way to go.
> 
> In my fork the iOS Demo app works, but the macOS Version needs some update, because it does not compile. Also I don't think that 3 different targets for iOS, macOS and tvOS are needed. This is the reason xcframeworks exists. I also have not tested if Cocoapods of SwiftPM works.

I need the xcframework from the latest svgkit (3.x branch), but I'm having difficulty with the conversion. How did you convert it to an xcframework?

**Comment by kjy95** [comment]
> I have created a fork and updated the project that xcframeworks can be build using carthage. I use this in one of my iOS project. Because there are some project changes that would break backwards compatibility with this project I have not yet created a pull request. If anybody is interested in the fork you can find it here: https://github.com/renep/SVGKit/tree/xcframework



> If I understand correctly: this is an overall improvement we should push ahead with.
> 
> Can you summarise the changes to backwards compatibility?
> 
> If they're big enough, but the advantage is enough, could be worth us moving to v4.x?

I need the xcframework from the latest svgkit (3.x branch), but I'm having difficulty with the conversion. How did you convert it to an xcframework?
